### PR TITLE
Documentation generation using sphinx and auto-push to gh-pages

### LINKED ISSUE #70: Implement documentation site using travis and sphinx
It seems to work out well with the site generated for the loklak documentation. 

Please have a look at the script (https://github.com/loklak/loklak_server/blob/development/.utility/push-docs-to-gh-pages.sh) and docs site and duplicate the way it is done here.

### PR DESCRIPTION
**Pull request by pri22296**
Fixes #70

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,5 @@ dist
 dist/loklak.jar
 /build/
 /.gradle/
+
+/docs/_build
```
Comment (yukiisbored) [inline .gitignore]: Missing new line on EOF

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ env:
   - secure: DbveaxDMtEP+/Er6ktKCP+P42uDU8xXWRBlVGaqVNU3muaRmmZtj8ngAARxfzY0f9amlJlCavqkEIAumQl9BYKPWIra28ylsLNbzAoCIi8alf9WLgddKwVWsTcZo9+UYocuY6UivJVkofycfFJ1blw/83dWMG0/TiW6s/SrwoDw=
 script:
 # The gh-pages updating script
-- 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then bash ./scripts/gh-pagesUpdater.sh; fi'
+#- 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then bash ./scripts/gh-pagesUpdater.sh; fi'
```
Comment (yukiisbored) [inline .travis.yml]: Just remove this line, No need to comment it

File: .utility/CNAME [hunk 3]
```diff
@@ -0,0 +1 @@
+dev.asksusi.com
```
Comment (yukiisbored) [inline .utility/CNAME]: Missing new line on EOF

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+The basis configuration file is in ``conf/config.properties``. To
+customize these settings place a file ``customized_config.properties``
+to the path ``data/settings/``
+
+How to compile using Gradle?
+----------------------------
+
+-  To install Gradle on Ubuntu:
+
+\`\`\` $ sudo add-apt-repository ppa:cwchien/gradle
```
Comment (yukiisbored) [inline README.rst]: Please fix this

File: docs/_static/images/cow.svg [hunk 5]
```diff
@@ -0,0 +1,191 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (yukiisbored) [inline docs/_static/images/cow.svg]: Please change this into SUSI's logo

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'alabaster'
+
+# Theme options are theme-specific and customize the look and feel of a theme
+# further.  For a list of options available for each theme, see the
+# documentation.
+html_theme_options = {
+    'logo': '/images/cow.svg',
```
Comment (yukiisbored) [inline docs/conf.py]: Change this to SUSI's logo

File: docs/installation/installation_generic.md [hunk 7]
```diff
@@ -1,23 +1,23 @@
-# Installation on Linux or Mac OS
+# Installation on Linux
 
 Its very easy!
 
-## Requirements
+##Requirements
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_generic.md [hunk 8]
```diff
@@ -1,23 +1,23 @@
...
 - install git, jdk 7 and ant
 
-## Download
+##Download
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_generic.md [hunk 9]
```diff
@@ -1,23 +1,23 @@
...
 - `git clone https://github.com/fossasia/susi_server.git`
 - `cd susi_server`
 
-## Build
+##Build
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_generic.md [hunk 10]
```diff
@@ -1,23 +1,23 @@
...
 - `ant` (just this, type "ant" - without quotes - and hit enter)
 
-## Run
+##Run
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_generic.md [hunk 11]
```diff
@@ -1,23 +1,23 @@
...
 - `bin/start.sh`
 
-## Operate
+##Operate
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_generic.md [hunk 12]
```diff
@@ -1,23 +1,23 @@
...
 - open `http://localhost:4000` in your browser
 
-## Shut down
+##Shut down
```
Comment (yukiisbored) [inline docs/installation/installation_generic.md]: Revert this line

File: docs/installation/installation_windows_generic.md [hunk 13]
```diff
@@ -7,7 +7,7 @@
 
 Install in terminal:
 
-#To be done as system variables: (Run as Admin)
+###To be done as system variables: (Run as Admin)
```
Comment (yukiisbored) [inline docs/installation/installation_windows_generic.md]: Please add a space between the heading syntax and the text

File: README.rst (hunk 4 above)
Comment (yukiisbored) [inline README.rst]: Use the same directive as https://github.com/fossasia/susi_server/pull/95/files/358044a2015697c8fa33d025f7ad47ea6951ac8a#diff-88b99bb28683bd5b7e3a204826ead112R47

**Comment by yukiisbored** [comment]
Don't forget to delete the original `README.md` because it's replaced by the new one

File: README.rst (hunk 4 above)
Comment (pri22296) [inline README.rst]: @yukiisbored Sorry, I could not understand what issue are you pointing out here. Could you please elaborate?

Comment (yukiisbored) [inline README.rst]: Please use the monospace/code syntax for this. The conversion isn't done properly.

**Comment by pri22296** [comment]
@yukiisbored I made those changes you suggested. Please review!

**Comment by yukiisbored** [comment]
There are still issues. Also, please rebase. I don't recommend using GitHub's resolve conflicts button.

**Comment by yukiisbored** [comment]
@pri22296 It looks like you didn't rebase it properly. Feel free to ask on gitter

**Comment by aviaryan** [comment]
@pri22296 There are conflicts in this PR. Please resolve them.

**Comment by pri22296** [comment]
@aviaryan I might have messed up the history while rebasing and now I am having trouble rebasing properly. What should i do?

**Comment by aviaryan** [comment]
@pri22296 What error log do you get while rebasing? Posting it here or on gitter might help. 
You can also send a fresh PR if nothing works out. 
Cheers 😄

**Comment by yukiisbored** [comment]
@pri22296 
* Make a backup of your old branch
* Destroy this branch
* Create a branch with the same name as this one but with more updated code
* Cherry-pick commits and fix conflicts
* Force Push

**Comment by yukiisbored** [comment]
@aviaryan The branch that GitHub sees is just a name, no need to create a new PR.

**Comment by aviaryan** [comment]
Just for everyone's information, PR #99 has been opened in place of it.

**Pull request closed without merging**